osu.Game/Overlays/OverlayScrollContainer.cs

```diff
@@ -0,0 +1,155 @@
+
+            protected override void PopOut() => button.FadeOut(fade_duration, Easing.OutQuint);
+
+            private class Button : OsuHoverContainer
+            {
+                public override bool PropagatePositionalInputSubTree => Alpha == 1;
+
+                protected override IEnumerable<Drawable> EffectTargets => new[] { background };
+
+                private Color4 flashColour;
+
+                private readonly Container content;
+                private readonly Box background;
+
+                public Button()
+                {
+                    RelativeSizeAxes = Axes.Both;
+                    Add(content = new CircularContainer
+                    {
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Masking = true,
+                        EdgeEffect = new EdgeEffectParameters
+                        {
+                            Type = EdgeEffectType.Shadow,
+                            Offset = new Vector2(0f, 1f),
+                            Radius = 3f,
+                            Colour = Color4.Black.Opacity(0.25f),
+                        },
+                        Children = new Drawable[]
+                        {
+                            background = new Box
+                            {
+                                RelativeSizeAxes = Axes.Both
+                            },
+                            new SpriteIcon
+                            {
+                                Anchor = Anchor.Centre,
+                                Origin = Anchor.Centre,
+                                Size = new Vector2(15),
+                                Icon = FontAwesome.Solid.ChevronUp
+                            }
+                        }
+                    });
+
+                    TooltipText = @"Scroll to top";
+                }
+
+                [BackgroundDependencyLoader]
+                private void load(OverlayColourProvider colourProvider)
+                {
+                    IdleColour = colourProvider.Background6;
+                    HoverColour = colourProvider.Background5;
+                    flashColour = colourProvider.Light1;
+                }
+
+                protected override bool OnClick(ClickEvent e)
+                {
+                    background.FlashColour(flashColour, 800, Easing.OutQuint);
+                    return base.OnClick(e);
+                }
+
+                protected override bool OnMouseDown(MouseDownEvent e)
+                {
+                    content.ScaleTo(0.75f, 2000, Easing.OutQuint);
+                    return base.OnMouseDown(e);
+                }
+
+                protected override void OnMouseUp(MouseUpEvent e)
+                {
+                    content.ScaleTo(1, 1000, Easing.OutElastic);
+                    base.OnMouseUp(e);
+                }
```

Comment:
Please explicitly return `true` or `false` as much as possible when overriding UI handlers. Now `OnClick` returns `true` as the base impl in `ClickableContainer` does so, but `OnMouse{Up,Down}` don't because they go down to `Drawable.HandleUIEvent`.

Reply:
We should call `base.OnClick` because action handled in there
https://github.com/ppy/osu-framework/blob/20cf599582d5ec2e9cd1bfe7231e8352ba7efe0c/osu.Framework/Graphics/Containers/ClickableContainer.cs#L26